### ISSUE TITLE
Multi valued parameters

### ISSUE THREAD

**Issue by MerlinSmiles** [QCoDeS/Qcodes#207]
I've been running into issues with multi valued parameters.

i.e. in the magnet driver, I want a `fld` parameter that returns and sets `x, y, z` values, reading the values works just fine with the `Anything` validator, but setting failes.

For reference, something like this:

``` Python
self.add_parameter('fld',
                   names=['B'+ax.lower() for ax in self.axes],
                   get_cmd=partial(self._get_fld, self.axes, 'FLD'),
                   set_cmd=partial(self._ramp_to_setpoint, self.axes, 'FSET'),
                   units=['T' for ax in self.axes],
                   vals=Numbers())
```

There are at least two issues here:
1. in the `parameter.py` file, lines `146+`:

``` Python
        if names is not None:
...
        elif name is not None:
...
        else:
            raise ValueError('either name or names is required')
```

The multi valued parameters no longer have a `name`, and a `_vals` attribute anymore.

If I do 

``` Python

        if name is not None:
...
        if names is not None:
...
        if (name is None) and (names is None):
```

It kindof works, but at least the docstring and units is broken (and I have to use the `Anything` validator)
2. The validators only work on single-valued inputs, not on multi-values.

**Comment by jenshnielsen** [comment]
related to #378, #166 and #498

**Comment by jenshnielsen** [comment]
I believe this was solved along with #166
